app/domain/content/importers/metrics.rb

```diff
@@ -0,0 +1,117 @@
+module Content
+  module Importers
+    class Metrics
+      def initialize(content_ids)
+        @content_ids = content_ids
+        @start_date = 3.months.ago.to_date
+      end
+
+      def import
+        start_time = Time.now
+        metrics = @content_ids
+          .map { |content_id| fetch_metrics(content_id) }
+          .flatten
+        fetched_metrics_end = Time.now
+        puts "Fetched all in #{fetched_metrics_end - start_time}s"
+
+        puts "Writing metrics to database..."
+        # TODO: on_duplicate_key_update
+        Content::Metric.import(metrics)
+        puts "Imported all in #{Time.now - fetched_metrics_end}s"
+      end
+
+    private
+
+      def fetch_metrics(content_id)
+        puts "Fetching metrics for #{content_id} ..."
+        versions = fetch_versions(content_id)
+
+        metrics_by_version = {}
+
+        metrics = (@start_date..Date.today).map do |date|
+          version = versions.find { |v| Date.parse(v[:updated_at]) <= date }
+
+          if version.nil?
+            nil
```

Comment:
What is the `nil` value for here? 

If `version == nil` then `version[:title].length` on line 42 could cause us some problems right?

Reply:
This `if...else` block is exactly here to stop problems with `version[:title]` (which is in the `else` bit). The `nil` is compacted out on line 54.

Reply:
Oops, I meant line 48 `version[:content_id]` which is outside of the `else` statement. :man_facepalming:

Reply:
Line 48 is also inside the `else` block. I admit there's an icky amount of nesting here. If this was production code I would definitely tidy it up.

Reply:
Ha oh `code reading fail`. Sorry!